spec/spec.md

```diff
@@ -0,0 +1,601 @@
+:::
+
+Further, the key manager interface ****MUST**** be passed as an argument to all public API methods that require key material such as:
+* DID Creation
+* Data Signing
+
+::: note
+[AWS KMS](https://docs.aws.amazon.com/kms/latest/developerguide/asymmetric-key-specs.html#key-spec-ecc) does not support [[ref:Ed25519]]. At some point in the future, we can consider introducing support for key wrapping.
+:::
+
+Consumers of conformant [[ref:Web5 SDKs]] ****SHOULD**** be able to provide their own `KeyManager` impelementations, if desired.
+
+### JSON Schema
+
+[[ref:JSON-SCHEMA]] support is ****REQUIRED**** for multiple features throughout [[ref:Web5 SDKs]], including, but not limited to support with [[ref:verifiable credentials]], [VC JSON Schemas](#vc-json-schema), and for use in validating the data models this specification has defined.
+
+Conformant implementations of [[ref:Web5 SDKs]] ****MUST**** support at least [[ref:JSON-SCHEMA-DRAFT-7]] and [[ref:JSON-SCHEMA-2020-12]].
+
+:::todo
+Elaborate on required features for JSON Schema support such as constructing and validating JSON Schemas. Provide guidance on caching and offline resolution.
+:::
+
+### Decentralized Identifiers (DIDs)
+
+Decentralized Identifiers (DIDs) are a core component of Web5, providing a foundation for self-sovereign identity and decentralized authentication, authorization, and discovery of services. Web5 implements a subset of the [[ref:DID-CORE]] specification to ensure consistency and interoperability across implementations. More detail can be found in the [section on the DID Document Data Model](#did-document-data-model).
+
+DIDs in Web5 are globally unique identifiers that enable verifiable, decentralized digital identity. They are designed to be independent of centralized registries, identity providers, or certificate authorities. Key features of [[ref:DIDs] in Web5 include:
+
+1. **Decentralization**: [[ref:DIDs]] can be created and managed without needing to rely on centralized authorities.
+2. **Control**: The controller has full say over their [[ref:DID]] and associated [[ref:DID Document]].
+3. **Persistence**: [[ref:DIDs]] are persistent identifiers that do not require the continued operation of an underlying organization.
+4. **Resolvability**: [[ref:DIDs]] can be resolved to retrieve associated [[ref:DID Documents]] containing cryptographic material, service endpoints, and other data.
+4. **Cryptographic Verifiability**: [[ref:DIDs]] enable cryptographic verification of claims and interactions.
+
+#### Supported DID Methods
+
+Web5 supports the following DID methods:
+
+| Method | Creation | Resolution | Note |
+|--------|----------|------------|------|
+| [[ref:did:web]] | ❌ | ✅ | - |
+| [[ref:did:jwk]] | ✅ | ✅ | - |
+| [[ref:did:dht]] | ✅ | ✅ | This is the "default" method for [[ref:Web5 SDKs]]. |
+| [[ref:did:key]] | ⚠️ | ⚠️ | This method has been implemented in both Kotlin and TypeScript, with no plans for support in other languages. |
```

Comment:
I would probably just remove this line altogether

The new KT SDK (from Rust Core) won't have this